storage/backend_key.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"encoding/binary"
+
+	"github.com/coreos/etcd/storage/storagepb"
+)
+
+// backendKeyBytesLen is the length of backendKey bytes.
+// First a few bytes stores the revision of backendKey. The following byte
+// is a ' ' as delimiter. The last 4 bytes is the event type in big-endian
+// format.
+const backendKeyBytesLen = revBytesLen + 1 + 4
+
+type backendKey struct {
+	rev revision
+	t   storagepb.Event_EventType
```

Comment:
@yichengq So we want to remove the watch event concept from the kv layer since I feel this is not necessary as I told you. 

There are multiple ways:

1. having another structure around the original kv to have tombstone information

2. embed the tombstone into the key space

They will both work. It seems 2 is simpler though. Again the core idea is to remove the event concept.